connectors/sources/google_drive.py

```diff
@@ -0,0 +1,670 @@
+        a different account they don't count against the user storage quota and therefore have size 0.
+        They need to be exported to a supported file type before the content extraction phase.
+
+        Args:
+            blob (dict): Formatted blob document.
+            timestamp (timestamp, optional): Timestamp of blob last modified. Defaults to None.
+
+        Returns:
+            dict: Content document with id, timestamp & text
+        """
+
+        blob_name, blob_id, blob_mime_type = blob["name"], blob["id"], blob["mime_type"]
+
+        document = {
+            "_id": blob_id,
+            "_timestamp": blob["_timestamp"],
+        }
+
+        attachment, blob_size = await self._download_content(
+            blob=blob,
+            download_func=partial(
+                self._google_drive_client.api_call,
+                resource="files",
+                method="export",
+                fileId=blob_id,
+                mimeType=GOOGLE_MIME_TYPES_MAPPING[blob_mime_type],
+            ),
+        )
+
+        # We need to do sanity size after downloading the file because:
+        # 1. We use files/export endpoint which converts large media-rich google slides/docs
+        #    into text/plain format. We usually we end up with tiny .txt files.
+        # 2. Google will ofter report the Google Workspace shared documents to have size 0
+        #    as they don't count against user's storage quota.
+        if blob_size > FILE_SIZE_LIMIT:
+            self._logger.warning(
+                f"File size {blob_size} of file {blob_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding the file content"
+            )
+            return
+
+        document["_attachment"] = attachment
+        return document
+
+    async def get_generic_file_content(self, blob, timestamp=None):
+        """Extracts the content from allowed file types supported by Apache Tika.
+
+        Args:
+            blob (dict): Formatted blob document.
+            timestamp (timestamp, optional): Timestamp of blob last modified. Defaults to None.
+
+        Returns:
+            dict: Content document with id, timestamp & text
+        """
+
+        blob_size = int(blob["size"])
+
+        if blob_size == 0:
+            return
+
+        blob_name, blob_id, blob_extension = (
+            blob["name"],
+            blob["id"],
+            f".{blob['file_extension']}",
+        )
+
+        if blob_extension not in TIKA_SUPPORTED_FILETYPES:
+            self._logger.debug(f"{blob_name} can't be extracted")
+            return
+
+        if blob_size > FILE_SIZE_LIMIT:
+            self._logger.warning(
+                f"File size {blob_size} of file {blob_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding the file content"
+            )
+            return
+
+        document = {
+            "_id": blob_id,
+            "_timestamp": blob["_timestamp"],
+        }
```

Comment:
This will actually be removed here:
https://github.com/elastic/connectors-python/blob/7fbb60ea586c95af4419659b8c25e28d5d022ac2/connectors/es/sink.py#L327-L328

I see the same pattern in other source implementations. Is it intended? cc @elastic/ingestion-team

Reply:
Yeah afaik it's intended - @tarekziade originally implemented it this way. 

We pop `ids` always because `id` is not a part of the document - it's a part of document metadata. Why `_timestamp` is popped I don't remember. Let's create an issue to address it?

Reply:
https://github.com/elastic/enterprise-search-team/issues/5095

Reply:
What I meant, it's not necessary to include `_id` and `_timestamp` here. It will eventually be removed from here and not used at all:
https://github.com/elastic/connectors-python/blob/7fbb60ea586c95af4419659b8c25e28d5d022ac2/connectors/es/sink.py#L327-L328

For the `lazy_download` function, it should be enough to just return a document with key `_attachment`.

But this should not block this PR.